rust-runtime/aws-smithy-http/src/query_writer.rs

```diff
@@ -3,19 +3,19 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-use crate::http_request::url_escape::percent_encode_query;
+use crate::query::fmt_string as percent_encode_query;
 use http::Uri;
 
 /// Utility for updating the query string in a [`Uri`].
-pub(super) struct QueryWriter {
+pub struct QueryWriter {
```

Comment:
Since this is becoming public, I think we should make it `#[doc(hidden)]` to indicate it is not intended to be used directly (it is an internal utility for codegen only, and the flexibility to change it at any time is definitely desired).